docs/source/concepts/interoperability.rst

```diff
@@ -0,0 +1,464 @@
+  Let `G` be an interoperable consistent causality graph and `Y` be a set of DAML ledgers.
+  The **projection** of `G` to party `P` on `Y` (`P`\ -**projection** on `Y`) is the `P`\ -reduction on `Y` of the following causality graph `G'`, which is `P`\ -consistent on `Y`:
+
+  * The vertices of `G'` are the vertices of `G` projected to `P` on `Y`, excluding empty projections.
+
+  * There is an edge between two vertices `v`:sub:`1` and `v`:sub:`2` in `G'` if there is an edge from the `G`\ -vertex corresponding to `v`:sub:`1` to the `G`\ -vertex corresponding to `v`:sub:`2`.
+
+If `G` is an interoperable consistent causality graph, then the `P`\ -projection on `Y` is `P`\ -consistent on `Y`, too.
+
+For example, the :ref:`interoperable causality graph for the paint counteroffer workflow <counteroffer-interoperable-causality-graph>` is projected as follows.
+
+.. https://app.lucidchart.com/documents/edit/d788b464-d670-4029-b2c0-d537c023052f
+   
+.. image:: ./images/counteroffer-causality-ledgeraware-projection.svg
+   :align: center
+   :width: 100%
+
+The following points are worth highlighting:
+
+* In Alice's projection on the green ledger, Alice witnesses the archival of her `Iou`.
+  As explained in the :ref:`interop-ordering-guarantees` below,
+  the **Exercise** action is marked as merely being witnessed
+  in the transaction stream of a Participant Node that hosts Alice on the green ledger but not on the yellow ledger.
+  Similarly, the Painter merely witnesses the **Create** of his `Iou` in the Painter's projection on the green ledger.
+
+* In the Painter's projections, the `ShouIou` transaction `tx3` is unordered w.r.t. to the `CounterOffer` acceptance in `tx4`
+  like in the :ref:`case of ordinary causality graphs <counteroffer-causality-projections>`.
+  The edge `tx3` -> `tx4` is removed by the reduction step during projection.
+
+The projection of transfer actions can be illustrated with the :ref:`interoperable-causality-graph-linear`.
+The `A`-projections on the yellow and green ledger look as follows.
+The white color indicates that a transfer action has no incoming or outgoing ledger annotation.
+That is, a **Leave** action is white on the right hand side and an **Enter** action is white on the left hand side.
+
+.. https://app.lucidchart.com/documents/edit/edbf9aaf-b7da-4e68-b9c9-9e631c3a87bb
+
+.. image:: ./images/transfer-projection.svg
+   :align: center
+   :width: 100%
+
+.. _interop-ordering-guarantees:
+
+Ledger API ordering guarantees
+******************************
+
+The Transaction Service and the Active Contract Service are derived from the local ledger that the Participant Node maintains for the party.
+Let `Y` be the set of ledgers on which the Participant Node hosts a party.
+The transaction tree stream outputs a topological sort of the party's local ledger on `Y`, with the following modifications:
+
+#. **Transfer** actions with both incoming and outgoing ledger annotations are omitted.
```

Comment:
So the Transaction Tree service doesn't contain transfer information, but the flat stream does? That seems odd to me. The flat service should surely be a projection of the tree one.

Reply:
The transaction tree service does contain the `Enter` and `Leave` events, just like the flat service. Added a sentence to make this clear.